Contents/Code/__init__.py

```diff
@@ -126,58 +206,90 @@ def Update(self, metadata, media, lang, force, movie):
         flags = flags | Prefs['hideSpoilerTags']    << 4 #0b10000 : Hide Plot Spoiler Tags
 
         if movie:
-            series = HttpReq("api/serie/fromep?id=%s&level=3&allpics=1&tagfilter=%d" % (aid, flags))
-            movie_episode_data = HttpReq("api/ep?id=%s" % (aid))
-            aid = try_get(series, 'id', None)
-            if not aid:
-                Log('Error! Series not found.')
-                return
-
-            if movie_episode_data['name'] == 'Complete Movie':
-                movie_name = series['name']
-                movie_sort_name = series['name']
+            # Get series data
+            series_data = {}
+            series_data['shoko'] = HttpReq('api/v3/Episode/%s/Series' % aid) # http://127.0.0.1:8111/api/v3/Series/24
+            series_id = series_data['shoko']['IDs']['ID']
```

Comment:
Needs to be more error resistant

Reply:
We are searching by episode ID linked to the file. I don't see how it cannot have a series ID associated with it if it has an episode ID.
But if that can happen somehow, I'll add a check